src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContextHelpers.cs

```diff
@@ -91,6 +94,7 @@ private static void VerifyDiagnosticLocationInCompilation(string id, Location lo
                 return;
             }
 
+            Debug.Assert(location.SourceTree != null);
```

Comment:
💡 Remove this assertion, and use the new member not null attribute on `IsInSource` to indicate that `SourceTree` will not be null when it's true.

Reply:
I think this will have much wider impact - we use `IsInSource` and `SourceTree` at bunch of places in the compiler. I am going to defer this to a separate issue for follow-up.

Reply:
https://github.com/dotnet/roslyn/issues/46729